mlperf_logging/result_summarizer/result_summarizer.py

```diff
@@ -521,23 +521,35 @@ def _compute_power_node(loglines, time_to_train):
     power_stop = 0
     agg_power = 0
     conversion_eff = 1.0
+    power_stop_found = False
+    power_start_found = False
+    loglines.sort(key=lambda x: x.timestamp)
     for logline in loglines:
         if logline.key == "power_measurement_start":
             power_start = logline.timestamp
             prev_timestamp = logline.timestamp
-        if logline.key == "power_reading":
-            agg_power += (logline.value['value'] * (logline.timestamp - prev_timestamp))
+            power_start_found = True
+        if (
+            (logline.key == "power_reading")
+            and (not power_stop_found)
+            and (power_start_found)
+        ):
+            agg_power += logline.value["value"] * (logline.timestamp - prev_timestamp)
             prev_timestamp = logline.timestamp
             max_timestamp = max(max_timestamp, logline.timestamp)
         if logline.key == "power_measurement_stop":
             power_stop = logline.timestamp
-            break
+            power_stop_found = True
         if logline.key == "conversion_eff":
-            conversion_eff = logline.value['value']
-    
+            conversion_eff = logline.value["value"]
+
+    # If power stop is not found, set it to the latest power reading
+    if not power_stop_found:
```

Comment:
is it expected that submission may not have the power-stop line? if not, maybe we should raise an error or log a warning?

Reply:
It is not expected. But it it happens, we can take the last power measurement as the stop. I updated it to log a warning

Reply:
https://github.com/mlcommons/logging/blob/f990bc8b037d031ef0a94188de4f6a468c6f8681/mlperf_logging/result_summarizer/result_summarizer.py#L550-L552

Reply:
https://github.com/mlcommons/logging/blob/640c988d1ee60123414f05a11fef91415ecdbe1e/mlperf_logging/result_summarizer/result_summarizer.py#L549-L552